### DISCUSSION TITLE
Help converting the following command to Magick.NET

### DISCUSSION THREAD

**Discussion by joseph-enriquez-messer-us** [ImageMagick/ImageMagick#6735] [Help]
### ImageMagick version

7.1.1-17

### Operating system, version and so on

Windows Server 2019

### Description

magick.exe convert 1.png -crop 504x720 +repage pdfa:1.pdf

The image is 344x1418px. Trying to crop it to 504x720 into multiple PDF pages.

**Comment by fmw42** [comment]
On IM 7 use magick, not magick convert.  However, that likely won't matter. I do not see anything obviously wrong to me, if IM supports PDFA.  I am not sure it does.  

If it is not listed on your system as rw+ for PDFA when you do 

`magick.exe -list format`

then it does not.

**Reply by joseph-enriquez-messer-us** [reply]
Thanks for the quick response. Sorry if I was not clear. I was actually looking for the .NET conversion of the command. I know the command works fine as it is currently running in Production. PDFA is the least of my concerns at the moment.

**Reply by fmw42** [reply]
So all you want is to crop an image in Magick.NET?  One of the IM developers will need to help you to do that, if that is all you want.

**Reply by joseph-enriquez-messer-us** [reply]
Yes, cropping one image that is very long (e.g. supermarket/costco receipt) to a specific size (504x720) then saving the cropped images into one PDF. Each cropped image is saved as a page in the PDF. First PDF page is cropped image 1, second PDF page is cropped imaged 2, etc. I would greatly appreciate it if you can send this conversation to one of the IM developers. I'm not sure if they see this conversation. Thank you.

**Reply by fmw42** [reply]
They see all questions. But they are very busy and may not answer quickly.

**Reply by dlemstra** [reply]
We see all the questions but we also have a personal life so it might take a while before we have time to answer questions. 

Most of the commands are methods on the `MagickImage` class. But yours is a but of a special case. Your command would translate to this:

```C#
using var image = new MagickImage("1.png");
using var images = new MagickImageCollection(image.CropToTiles(504, 720));
images.Write("1.pdf", MagickFormat.Pdfa);
```

**Reply by joseph-enriquez-messer-us** [reply]
Sorry I didn't mean to rush everyone. It wasn't my intention. This is my first time posting so I'm still learning a lot. At the moment, I can generate the PDF from a PNG but the whole image is only in one page. I will try the code you posted and see the results.  Once again, thank you.

**Reply by dlemstra** [reply]
No worries just wanted to give an explanation why it took a while before you got a response 😁

**Reply by joseph-enriquez-messer-us** [reply]
Success!!! Thank you so much for pointing me in the right direction. I spent a day looking at samples I could find online and of course the documentation. It's such a powerful tool and there's so many things you can do with it. I think I confused myself trying different things. Using the suggested code above, it looked like the image was cut in the middle and added a lot of blank white space at the top and at the bottom (where the original image didn't have any of those blank space). The RePage removed those blank spaces. Anyways, below is what worked for me.

`using var image = new MagickImage("input.png");
using var images = new MagickImageCollection(image.CropToTiles(504, 720));
foreach (var img in images)
{
	img.RePage();
}
images.Write("output.pdf");`

**Reply by joseph-enriquez-messer-us** [reply]
Oh yeah, I forgot to put in PDFA but that's now the easy part. 😊
